models/lpjguess/R/met2model.LPJGUESS.R

```diff
@@ -0,0 +1,140 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2012 University of Illinois, NCSA.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the 
+# University of Illinois/NCSA Open Source License
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+#
+## R Code to convert NetCDF CF met files into LPJ-GUESS met files
+
+##If files already exist in "Outfolder", the default function is NOT to overwrite them and 
+##only gives user the notice that file already exists. If user wants to overwrite the existing files, just change 
+##overwrite statement below to TRUE.
+
+##' met2model wrapper for LPJ-GUESS
+##'
+##' @title met2model.LPJGUESS
+##' @export
+##' @param in.path location on disk where inputs are stored
+##' @param in.prefix prefix of input and output files
+##' @param outfolder location on disk where outputs will be stored
+##' @param start_date the start date of the data to be downloaded (will only use the year part of the date)
+##' @param end_date the end date of the data to be downloaded (will only use the year part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose should the function be very verbose
+##' @author Istem Fer
+met2model.LPJGUESS <- function(in.path, in.prefix, outfolder, start_date, end_date, ..., overwrite=FALSE,verbose=FALSE){
+  
+  library(PEcAn.utils)
+  require(ncdf4)
+  require(lubridate)
+  
+  print("START met2model.LPJGUESS")
+  start_date <- as.POSIXlt(start_date, tz = "GMT")
+  end_date <- as.POSIXlt(end_date, tz = "GMT")
+  start_year <- year(start_date)
+  end_year <- year(end_date)
+  
+  year = sprintf("%04d",seq(start_year,end_year,1))
+  nyear = length(year) #number of years to simulate
+  
+  ## LPJ-GUESS looks for different input files for different climate variables
+  out.file <- out.files.full <- list()
+  var.names=c("tmp","pre","cld")
+  n.var=length(var.names)
+  long.names=c("air_temperature","precipitation_flux","surface_downwelling_shortwave_flux_in_air")
+  for(i in 1:n.var) out.file[[i]] <-paste(in.prefix, start_year, end_year, var.names[[i]], "nc", sep=".")
+  for(i in 1:n.var) out.files.full[[i]] <-file.path(outfolder,out.file[[i]])
+  
+  results <- data.frame(file = unlist(out.files.full),
+                        host = fqdn(),
+                        mimetype ='application/x-netcdf',
+                        formatname = 'lpj-guess.metfile' ,
+                        startdate = start_date ,
+                        enddate = end_date,
+                        dbfile.name = unlist(out.file),
```

Comment:
In principle it should be fine. eg you could pass a directory name where the three files are. Or in BioCro names of met and soil input files are given separately.